### PR TITLE
Enhancements to the nursing care procedures and routines tables

### LINKED ISSUE #8562: Enhancements to Nursing tab UI
### Steps to Reproduce:

1. Go to a consultation
2. Create a log update (ensure bed is assigned) of type "Community Nurses Log"
3. Fill routine and nursing care section
4. Once submitted, check the Nursing Tab in the consultation dashboard

image

### Action Items

- [ ] lock the first row of the field name in routine section table
- [ ] have fixed width for time columns (2nd column and after)
- [ ] Make the Nursing Care section similar to the Routine Section by switching to a table layout (Make the component a reusable component to achieve it).
- [ ] Optional: feel free to improve the overall UI (borders or colors, etc.)

_Originally posted by @nihal467 in https://github.com/ohcnetwork/care_fe/issues/8522#issuecomment-2358037877_

### PR DESCRIPTION
**Pull request by sainak**
Picked up changes from https://github.com/ohcnetwork/care_fe/pull/8608

Changes made
fixes: https://github.com/ohcnetwork/care_fe/issues/8562
1. Added a reusable component.
2. fixed width of the 2nd column onwards

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `NursingPlot` component for enhanced display of nursing round data.
	- Added the `LogUpdateAnalyseTable` component for improved rendering of routine analysis results.

- **Improvements**
	- Streamlined data-fetching logic and pagination in the `ConsultationNursingTab`.
	- Enhanced type safety for nursing data structure.

- **Bug Fixes**
	- Improved handling of empty fields in nursing care procedures.

### CODE REVIEW

**Comment by sainak** [comment]
@coderabbitai full review

**Comment by sainak** [comment]
@coderabbitai full review

File: src/components/Facility/ConsultationDetails/ConsultationNursingTab.tsx [hunk 1]
```diff
@@ -114,6 +91,104 @@
...
+    ) => {
+      const { res, data } = await request(routes.dailyRoundsAnalyse, {
+        body: { page: currentPage, fields: NursingPlotFields },
+        pathParams: { consultationId },
+      });
+      if (res && res.ok && data) {
+        setResults(data.results);
+        setTotalCount(data.count);
+      }
+    };
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/components/Facility/ConsultationDetails/ConsultationNursingTab.tsx]: the request function takes care of showing error notification

File: src/components/Facility/ConsultationDetails/ConsultationNursingTab.tsx [hunk 2]
```diff
@@ -114,6 +90,94 @@ const ROUTINE_ROWS = [
...
+        pathParams: { consultationId },
+      });
+      if (res?.ok && data) {
+        setResults(data.results as { [date: string]: NursingPlotRes });
+        setTotalCount(data.count);
+      }
+    };
+
+    fetchDailyRounds(currentPage, consultationId);
+  }, [consultationId, currentPage]);
```
[the remark this answers is not in the document]
Comment (sainak) [reply src/components/Facility/ConsultationDetails/ConsultationNursingTab.tsx:117]: the parent component already shows a loading indicator

File: src/components/Facility/ConsultationDetails/ConsultationNursingTab.tsx [hunk 3]
```diff
@@ -4,43 +4,20 @@ import { useTranslation } from "react-i18next";
 import Loading from "@/components/Common/Loading";
 import PageTitle from "@/components/Common/PageTitle";
 import Pagination from "@/components/Common/Pagination";
+import { ProcedureType } from "@/components/Common/prescription-builder/ProcedureBuilder";
 import { ConsultationTabProps } from "@/components/Facility/ConsultationDetails/index";
-import { NursingPlot } from "@/components/Facility/Consultations/NursingPlot";
+import LogUpdateAnalayseTable from "@/components/Facility/Consultations/LogUpdateAnalayseTable";
```
[the remark this answers is not in the document]
Comment (rithviknishad) [reply src/components/Facility/ConsultationDetails/ConsultationNursingTab.tsx]: @sainak this typo suggestion is valid

**Review by rithviknishad — APPROVED** [review]
Other than https://github.com/ohcnetwork/care_fe/pull/9079/files#r1836226214 LGTM;

**Review by shivankacker — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**